src/vm/win32threadpool.h

```diff
@@ -1132,8 +1141,9 @@ class ThreadpoolMgr
         if (CLRThreadpoolHosted())
             return false;
 
-        DWORD requiredInterval = NextCompletedWorkRequestsTime - PriorCompletedWorkRequestsTime;
-        DWORD elapsedInterval = GetTickCount() - PriorCompletedWorkRequestsTime;
+        DWORD priorTime = PriorCompletedWorkRequestsTime; 
```

Comment:
In terms of correctness, I don't think it matters what it reads, I believe it's just a matter of avoiding unnecessary calls to AdjustMaxWorkersActive(). I think this change and the corresponding change to the stores in win32threadpool.cpp, could theoretically cause unnecessary calls to AdjustMaxWorkersActive(), though I don't think it's incorrect.

Before, we had something like this:

```
Stores        | Loads
-----------------------------
Prior = ...   | ... = Next
MemoryBarrier | MemoryBarrier
Next = ...    | ... = Prior
```

This makes it such that after a store of Next, the next load of Prior will be an updated value. It's possible for this to read an old Next and a new Prior, such that Prior > Next, but that seems to be ok since the required interval would then typically be very large, avoiding a call to AdjustMaxWorkersActive() in that case.

After your change, we have something like this:

```
Stores        | Loads
-----------------------------
MemoryBarrier | ... = Prior
Prior = ...   | ... = Next
Next = ...    | MemoryBarrier
```

With the memory barrier before stores and after loads, the loads may read older values, causing unnecessary calls to AdjustMaxWorkersActive().

I think it's unlikely for this to actually read old values, so it probably won't make any difference. But if you want to have only one memory barrier here, I suggest the following:

```
Stores        | Loads
-----------------------------
Next = ...    | ... = Prior
MemoryBarrier | MemoryBarrier
Prior = ...   | ... = Next
```

This prevents (Next - Prior) from overflowing (except when GetTickCount() overflows, which doesn't matter here), and (currentTime - Prior) can be usefully compared to (Next - Prior), so the comparison is always meaningful. It also prevents old reads from causing requiredInterval to decrease, it may only increase.